agent/agent.go

```diff
@@ -109,7 +111,7 @@ func (a *Agent) Leave(ctx context.Context) error {
 	case <-a.closed:
 		return ErrClosed
 	case <-ctx.Done():
-		return ctx.Err()
+		return errors.Wrap(ctx.Err(), "context done while waiting for agent to leave")
```

Comment:
waiting for agent to leave => waiting for agent to leave the cluster, or
waiting for agent.stop()